Source/WebKit/UIProcess/WebAuthentication/Cocoa/WebAuthenticatorCoordinatorProxy.mm

```diff
@@ -196,7 +197,14 @@ - (void)authorizationController:(ASAuthorizationController *)controller didCompl
         }
     }
     if (includePlatformRequest) {
-        auto request = adoptNS([[allocASAuthorizationPlatformPublicKeyCredentialProviderInstance() initWithRelyingPartyIdentifier:*options.rp.id] createCredentialRegistrationRequestWithChallenge:toNSData(options.challenge).get() name:options.user.name userID:toNSData(options.user.id).get()]);
+        RetainPtr<ASAuthorizationPlatformPublicKeyCredentialRegistrationRequest> request;
+        ASAuthorizationPlatformPublicKeyCredentialProvider *provider = [allocASAuthorizationPlatformPublicKeyCredentialProviderInstance() initWithRelyingPartyIdentifier:*options.rp.id];
+        if ([provider respondsToSelector:@selector(createCredentialRegistrationRequestWithClientData:name:userID:)]) {
+            ASPublicKeyCredentialClientData *clientData = [allocASPublicKeyCredentialClientDataInstance() initWithChallenge:toNSData(options.challenge).get() origin:origin];
```

Comment:
Should use a RetainPtr and a adoptNS() to avoid leaking.

Reply:
Done

Reply:
This doesn't look done? Still no RetainPtr or adoptNS().

Reply:
I don't know what happened, I swear I wasn't just lying :(
It should really be done this time.

Reply:
No worries, it happens.